teuthology/test/test_reimage_error_mark_machine_down.py

```diff
@@ -0,0 +1,75 @@
+                                                m_urljoin, m_requests):
+        with patch("teuthology.lock.ops.update_lock") as mark_down:
+            targets = {'fakeos@rmachine061.front.sepia.ceph.com': 'ssh-ed25519'}
+            m_requests.get.return_value.json.return_value = \
+                        self.create_n_out_of_10_reimage_failed_jobs(10)
+            self.the_function(targets)
+            self.assertTrue(mark_down.called)
+
+    @patch('teuthology.dispatcher.supervisor.requests')
+    @patch('teuthology.dispatcher.supervisor.urljoin')
+    @patch('teuthology.dispatcher.supervisor.teuth_config')
+    def test_one_machine_seven_reimage_failed_jobs(self, m_t_config,
+                                                    m_urljoin, m_requests):
+        with patch("teuthology.lock.ops.update_lock") as mark_down:
+            targets = {'fakeos@rmachine061.front.sepia.ceph.com': 'ssh-ed25519'}
+            m_requests.get.return_value.json.return_value = \
+                        self.create_n_out_of_10_reimage_failed_jobs(7)
+            self.the_function(targets)
+            self.assertFalse(mark_down.called)
+    
+    @patch('teuthology.dispatcher.supervisor.requests')
+    @patch('teuthology.dispatcher.supervisor.urljoin')
+    @patch('teuthology.dispatcher.supervisor.teuth_config')
+    def test_two_machine_all_reimage_failed_jobs(self, m_t_config,
+                                                m_urljoin, m_requests):
+        with patch("teuthology.lock.ops.update_lock") as mark_down:
+            targets = {'fakeos@rmachine061.front.sepia.ceph.com': 'ssh-ed25519',
+                       'fakeos@rmachine179.back.sepia.ceph.com': 'ssh-ed45333'
+                      }
+            m_requests.get.return_value.json.side_effect = [
+                        self.create_n_out_of_10_reimage_failed_jobs(10),
+                        self.create_n_out_of_10_reimage_failed_jobs(10)
+                        ]
+            self.the_function(targets)
+            self.assertEqual(mark_down.call_count, 2)
+
+    @patch('teuthology.dispatcher.supervisor.requests')
+    @patch('teuthology.dispatcher.supervisor.urljoin')
+    @patch('teuthology.dispatcher.supervisor.teuth_config')
+    def test_two_machine_one_healthy_one_reimage_failure(self, m_t_config,
+                                                        m_urljoin, m_requests):
+        with patch("teuthology.lock.ops.update_lock") as mark_down:
+            targets = {'fakeos@rmachine061.front.sepia.ceph.com': 'ssh-ed25519',
+                       'fakeos@rmachine179.back.sepia.ceph.com': 'ssh-ed45333'
+                      }
+            m_requests.get.return_value.json.side_effect = [
+                        self.create_n_out_of_10_reimage_failed_jobs(0),
+                        self.create_n_out_of_10_reimage_failed_jobs(10)
+                        ]
+            self.the_function(targets)
+            self.assertEqual(mark_down.call_count, 1)
```

Comment:
For this test we should also verify the _correct_ machine was marked down. With `py.test` it's as simple as:
`assert mark_down.call_args_list[0][0][0].startswith('rmachine179')`